Update README now that Wild Web Developer supports XML LS extensions #4

AObuchow opened the pull request:
Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow commented:
Note that the "Eclipse IDE Context" could also be updated (but it's not entirely necessary). 
I also wrote "XML-LS" rather than LSP4XML, if you want me to change that just let me know!

angelozerr commented:
@AObuchow I think you should change https://github.com/angelozerr/lsp4xml-extensions-maven#generate-proper-classpath

AObuchow committed:
WIP - Add Maven SubModule completion

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
WIP - Autocomplete in <parent> tag

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
WIP - Autocomplete in <parent> tag

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
WIP - Add diagnostic support, add artifact version diagnostics

Currently the artifact version validator does not work as intended.

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
WIP - Add diagnostic to check existence of submodule

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Remove duplicate maven dependencies in pom.xml

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Automated build support (create maven.yml)

Fix #1

AObuchow committed:
Add distributionManagement to pom.xml

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Add support for deploying jar to github packages

#1

AObuchow committed:
Update maven.yml

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Update maven.yml

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

mickaelistria committed:
Use own snapshot version

Initiate a specific versioning for the project as we're not sure it evolves and releases in sync with LSP4XML.
Also uses SNAPSHOT as we're not supposed to changed content for released version.

AObuchow committed:
Merge branch 'master' of github.com:AObuchow/lsp4xml-extensions-maven

AObuchow committed:
Merge pull request #2 from mickaelistria/patch-1

Use own snapshot version

mickaelistria committed:
Fix version for dependency

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #5 from mickaelistria/dep-version

Fix version for dependency

AObuchow committed:
Add distribution management for snapshots

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

mickaelistria committed:
License + Copyright headers

Fixes #4

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Remove parent

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Remove period's from artifactId

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Add CI for PR's

#6

AObuchow committed:
Merge pull request #8 from AObuchow/update_artifactId

Update artifactId

AObuchow committed:
Update Master CI name & trigger conditions

Rename Java CI to Master CI, make deploy build only occur for pushes to master

Fix #6

AObuchow committed:
Merge pull request #7 from mickaelistria/legal

License + Copyright headers

AObuchow committed:
Add M2E vs Generic Editor with Maven Extension Document #9

AObuchow committed:
Add M2E vs Generic Editor with Maven Extension Document #9

AObuchow committed:
Merge pull request #10 from AObuchow/add_m2e_vs_generic_editor_doc

Add M2E vs Generic Editor with Maven Extension Document

AObuchow committed:
Fix m2e vs generic editor doc format issue

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge pull request #11 from AObuchow/add_m2e_vs_generic_editor_doc

Fix m2e vs generic editor doc format issue

mickaelistria committed:
Add first test cases

Fixes #3

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Use Java 11 on PR CI

AObuchow committed:
Revert "Use Java 11 on PR CI"

This reverts commit 24c78b6890e09ed823bf94ff3b30d757e7875a7c.

AObuchow committed:
Merge pull request #14 from AObuchow/mickaelistria-tests

Add first test cases, fix #3

mickaelistria committed:
Report problems from ProjectBuilder as diagnostics

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #15 from mickaelistria/problem-diagnostics

Report problems from ProjectBuilder as diagnostics

AObuchow committed:
WIP - Add autocomplete for artifact latest version

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

mickaelistria committed:
Fix test name

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Introduce a MavenProjectCache

This cache stores the resolved Maven project and latest diagnostics, and
updates them when the document has changed.
Then operations can request the MavenProject from this cache.

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Fix RepositorySystem and access to local repo

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #17 from mickaelistria/cache

Add Maven Project Cache + Property completion + tests

AObuchow committed:
Fix version validator artifact initilization errors

Fix error on missing artifactId/groupId/version + add workingCopy files
to .gitignore

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

mickaelistria committed:
Fixes and refactoring to property completion

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #20 from AObuchow/Fix_version_validator_errors

Fix version validator artifact initilization errors

mickaelistria committed:
Try increasing timeout for tests on GH Actions

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #19 from mickaelistria/prop-completion

Fixes and refactoring to property completion

mickaelistria committed:
Support for "native" properties

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Read changed document instead of local file

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #22 from mickaelistria/fix-read

Read changed document instead of local file

mickaelistria committed:
Fix diagnostic line

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #23 from mickaelistria/native-props

Support for "native" properties autocomplete

AObuchow committed:
Version Bymp

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Version Bymp

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge pull request #25 from AObuchow/version_bump

Version bump to 0.0.2-SNAPSHOT

AObuchow committed:
Version autocomplete with Maven Indexer

Add MavenRepositoryCache, add ArtifactVersionSearcher which uses Maven
Indexer to search for artifact versions

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge pull request #24 from mickaelistria/diagnostic-line

Diagnostic line range fix

mickaelistria committed:
Add test for completion using parent properties

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #27 from mickaelistria/parent-props-test

Add test for completion using parent properties

mickaelistria committed:
Support for completion of <dependency>

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #28 from mickaelistria/dependency-completion

Support for completion of <dependency>

AObuchow committed:
Format codebase

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge pull request #42 from AObuchow/format_codebase

Format codebase

mickaelistria committed:
Add support for Phase completion

+ automatically close tag on completion
+ factorize generation of simple completion items from a list

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Enable verbose debug output for mavenPR.yml

mickaelistria committed:
Factorize methods related to local repo

+ Implement a watch on local repo to cache artifacts and refresh when
repo is modified

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Version autocomplete with Maven Indexer

Add MavenRepositoryCache, add ArtifactVersionSearcher which uses Maven
Indexer to search for artifact versions

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge branch 'master' into phase-completion

AObuchow committed:
Merge pull request #44 from mickaelistria/phase-completion

Add support for Phase completion

AObuchow committed:
Merge pull request #45 from mickaelistria/factorize-local-repo

Factorize methods related to local repo

AObuchow committed:
Remove debug flag from MavenPR.yml

AObuchow committed:
Merge pull request #26 from AObuchow/remote_version_autocomplete_indexer

Remote version autocomplete indexer

AObuchow committed:
Refactors in MavenCompletionParticipant, add test for local parent GAV
completion

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Remove "clean" flag from MavenPR.yml

AObuchow committed:
remove "package" flag from mavenPR.yml

AObuchow committed:
Merge pull request #46 from AObuchow/refactor_and_add_tests

Refactors in MavenCompletionParticipant, add test for local parent GAV completion

AObuchow committed:
Add remote groupId, artifactId completion


Refactors, add remote artifactId/groupId completions and tests for them,
rename ArtifactVersionSearcher to RemoteRepositoryIndexSearcher

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge pull request #49 from AObuchow/add_remote_groupid_artifactid_completion

Add remote groupId, artifactId completion

AObuchow committed:
WIP - Property evaluation on hoever

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

mickaelistria committed:
Refactoring

Refactor remote repository access to we get finer grain report about
which repos we're fetching index for, and results from one repo even if
others are pending for index download.

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Allow usage of incomplete pom models

Some projectBuilder error still allow for manipulation of the model; in
such case, store the "incomplete" model for further usage as it reflects
a more current state anyway.

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #51 from mickaelistria/incomplete-model

Allow usage of incomplete pom models

AObuchow committed:
Revert "WIP - Property evaluation on hoever"

This reverts commit 26774f8fa18e3e5df965e9003cae8a0b80186e42.

AObuchow committed:
Merge pull request #52 from mickaelistria/refactor-remote

Refactoring

AObuchow committed:
Move location of maven indexes

Fix #55

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge pull request #56 from AObuchow/move_maven_index_cache_location

Move location of maven indexes

AObuchow committed:
Add remote plugin GAV autocomplete, use model as fallback in
projectbuildingerror

GAV completion for #32

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge pull request #58 from AObuchow/maven_plugin_completion

Add remote plugin GAV autocomplete, use model as fallback in projectbuildingerror

mickaelistria committed:
Completion plugin for goal and configuration elts

This is already a bit valuable but not perfect. However, it doesn't hard
and provides interesting code to reuse, so it seems OK to merge.

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Maven plugin configuration & goal hover + refactorings

Signed-off-by: Andrew Obuchowicz aobuchow@redhat.com

AObuchow committed:
Merge pull request #59 from mickaelistria/complete-goal-configuration

Autocomplete Maven plugin goals & configuration

mickaelistria committed:
Improve configuration element completion

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Avoid creation of working copies

Use more flexible ModelSource2

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #60 from mickaelistria/improve-configuration-complete

Improve configuration element completion (support for onTagOpen)

mickaelistria committed:
Improve groupId completion

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #61 from mickaelistria/avoid-working-copy

Avoid creation of working copies

AObuchow committed:
Merge pull request #62 from AObuchow/maven_plugin_configuration_hover

Maven plugin configuration & goal hover + refactorings

AObuchow committed:
Merge pull request #63 from mickaelistria/improve-groupId

Improve groupId completion

mickaelistria committed:
Improve some completion

* Some code improvements
* Make completion on artifactId only complete if groupId match

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Improve some completion

* Some code improvements
* Make completion on artifactId only complete if groupId match
* Remove testLocalParentGAVCompletion

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Minimal pom generation

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Add artifactId description hover


Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Fix newlines in plugin configuration hover & completion description

Fix #71

Signed-off-by: Andrew Obuchowicz aobuchow@redhat.com

AObuchow committed:
Merge pull request #67 from mickaelistria/complete-artifactid-with-groupId

Improve some completions, check local repo for maven plugins, make completion on artifactId only complete if groupId match

AObuchow committed:
Merge pull request #70 from mickaelistria/minimal-pom

Minimal pom completion snippet

mickaelistria committed:
Support insertion of <groupId> and <version> with <artifactId>

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #72 from AObuchow/artifactId_description_hover

Add artifactId description hover

Fix #65

AObuchow committed:
Merge pull request #74 from AObuchow/fix_hover_completion_description_formatting

Fix newlines in plugin configuration hover & completion description

Fix #71

AObuchow committed:
Use \n\n instead of <br/> for getMarkupDescription

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Merge pull request #75 from AObuchow/fix_hover_completion_description_formatting

Use \n\n instead of <br/> for getMarkupDescription

Fix #71

mickaelistria committed:
Factorize code to insert only one GAV field or a whole GAV block

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Use org.eclipse.lemminx.maven package

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #73 from mickaelistria/insert-groupId-with-artifactId

Support insertion of <groupId> and <version> with <artifactId>

AObuchow committed:
Merge pull request #76 from mickaelistria/factorize-complete-gav-value-block

Factorize complete gav value block

AObuchow committed:
Merge branch 'rename-package' of
https://github.com/mickaelistria/lsp4xml-extensions-maven into
mickaelistria-rename-package

AObuchow committed:
Rename package

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
mend

AObuchow committed:
reference lemminx-maven in mavenPR.yml

AObuchow committed:
Merge branch 'master' into rename-package

AObuchow committed:
Merge pull request #79 from AObuchow/rename-package

Rename package

AObuchow committed:
reference lemminx-maven in mavenl.yml

mickaelistria committed:
Use same code to complete GAV at different levels

This also brings
* parent GAV completion when relative path is set
* completing inside `dependency` and `plugin`

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Factorize by using ArtifactInfo and deduplicate

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Factorize by using ArtifactInfo and deduplicate

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge pull request #80 from mickaelistria/insert-multi-level

Use same code to complete GAV at different levels, add parent completion

mickaelistria committed:
Use proper source selectionrange in definition

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Definition support

Link to file for
* module
* parent (when available locally)
* relativePath
* any GAV like dependency or plugin (when available locally)

Signed-off-by: Mickael Istria <mistria@redhat.com>

mickaelistria committed:
Use Eclipse Lemminx 0.11.0

Signed-off-by: Mickael Istria <mistria@redhat.com>

AObuchow committed:
Merge branch 'definition' of
https://github.com/mickaelistria/lsp4xml-extensions-maven into
mickaelistria-definition

# Conflicts:
#	lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenParseUtils.java
#	lemminx-maven/src/main/java/org/eclipse/lemminx/maven/searcher/LocalRepositorySearcher.java

AObuchow committed:
Bump version to 0.0.2-SNAPSHOT for m2e use

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Fix tests now that we're using LemMinX

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
update README to reference LemMinX in header

AObuchow committed:
Shutdown LS properly in tests

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Add maven property hover

Fix #30

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

AObuchow committed:
Update readme to reference new repo
